src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs

```diff
@@ -335,9 +361,18 @@ private async Task ProcessSocketAsync(WebSocket socket)
                 socket.Abort();
 
                 // Cancel any pending flush so that we can quit
-                _application.Output.CancelPendingFlush();
+                if (_gracefulClose)
+                {
+                    _application.Output.CancelPendingFlush();
+                }
             }
         }
+
+        if (_useAck && !_gracefulClose)
+        {
+            UpdateConnectionPair();
+            await StartAsync(url, _webSocketMessageType == WebSocketMessageType.Binary ? TransferFormat.Binary : TransferFormat.Text, default).ConfigureAwait(false);
```

Comment:
We should retry the seamless reconnect attempts until we hit a configured limit. It probably makes sense to use an `IRetryPolicy` for this like we do for normal reconnects.